File: python/easy/prob_008_product_sum/solution.py
def product_sum(array, level = 1):
  total = 0

  for element in array:
    if type(element) is list:
      total += product_sum(element, level + 1)
    elif type(element) is int:
      total += element

  return total * level

print(product_sum([1,2,3,[4,5],6,[7,8,[9,10,-200]]]))

File: ruby/easy/prob_008_product_sum/solution.rb
def product_sum(array, level = 1)
  total = 0
  array.each do |element|
    if element.is_a?(Array)
      total += product_sum(element, level + 1)
    elsif element.is_a?(Integer)
      total += element
    end
  end

  total * level
end

puts product_sum([5,2,[7,-1],3,[6,[-13,8],4]])